### PR TITLE
LibraryWindow.vala: improve Saved State

### LINKED ISSUE #363: Window shrinks on restore
Open and close Music repeatedly. Notice that the window gets shorter and shorter over time

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/59983133-window-shrinks-on-restore?utm_campaign=plugin&utm_content=tracker%2F60766398&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F60766398&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by danrabbit**
Fixes #363 

Update maximize state when it changes instead of waiting until the window is destroyed

### CODE REVIEW

**Comment by BaptisteGelez** [comment]
DConf tells me the state is correctly saved, the problem is restoring it.

**Comment by danrabbit** [comment]
Hm I definitely can't confirm that issue. I'm running on a fairly clean install of the latest Juno stable

**Comment by BaptisteGelez** [comment]
Maybe it's an issue with my setup, I'm still on Loki. The `set_allocation` behavior may have changed.

File: src/LibraryWindow.vala [hunk 1]
```diff
@@ -452,7 +452,13 @@ public class Noise.LibraryWindow : LibraryWindowInterface, Gtk.Window {
         icon_name = "multimedia-audio-player";
         title = _("Music");
 
-        set_default_size (App.saved_state.get_int ("window-width"), App.saved_state.get_int ("window-height"));
+        var window_height = App.saved_state.get_int ("window-height");
+        var window_width = App.saved_state.get_int ("window-width");
+        var rect = Gtk.Allocation ();
+        rect.height = window_height;
+        rect.width = window_width;
+        set_allocation (rect);
```
Comment (jeremypw) [inline src/LibraryWindow.vala]: Documentation says that `set allocation ()` should only be used inside a widget's `size_allocate ()` method.  But I am not sure what is wrong with set_default_size () for a window anyway?

**Pull request closed without merging**